core/src/main/java/org/wso2/carbon/kernel/utils/ConfigUtil.java

```diff
@@ -0,0 +1,613 @@
+     * @return String which has the new value instead of the placeholder.
+     */
+    private static String processValue(Function<String, String> func, String key, String inputString, String
+            defaultValue, Placeholder type) {
+        String newValue = func.apply(key);
+        //If the new value is not null, replace the placeholder with the new value and return the string.
+        if (newValue != null) {
+            return inputString.replaceFirst(PLACEHOLDER_REGEX, "$1" + newValue + "$8");
+        }
+        //If the new value is empty and the default value is not empty, replace the placeholder with the default
+        // value and return the string
+        if (defaultValue != null) {
+            return inputString.replaceFirst(PLACEHOLDER_REGEX, "$1" + defaultValue + "$8");
+        }
+        //Otherwise print an error message and throw na exception
+        String msg;
+        if (Placeholder.ENV.getValue().equals(type.getValue())) {
+            msg = String.format("Environment variable %s not found. Processing %s failed. Placeholder: %s", key,
+                    CONFIG_FILE_NAME, inputString);
+        } else if (Placeholder.SYS.getValue().equals(type.getValue())) {
+            msg = String.format("System property %s not found. Processing %s failed. Placeholder: %s", key,
+                    CONFIG_FILE_NAME, inputString);
+        } else {
+            msg = String.format("Unsupported placeholder type: %s", type.getValue());
+        }
+        logger.error(msg);
+        throw new RuntimeException(msg);
+    }
+
+    /**
+     * This method will concatenate and return the file types that need a root element. File types will be separated
+     * by | token. This method will be used to create the {@link ConfigUtil#FILE_REGEX}.
+     *
+     * @return String that contains file types which are separated by | token.
+     */
+    private static String getFileTypesString() {
+        StringBuilder stringBuilder = new StringBuilder();
+        for (FileType fileType : FileType.values()) {
+            stringBuilder.append(fileType.getValue()).append("|");
+        }
+        String value = stringBuilder.toString();
+        logger.debug("FileTypes String: {}", value);
+        return value.substring(0, value.length() - 1);
+    }
+
+    /**
+     * This method will concatenate and return the placeholder types.. Placeholder types will be separated
+     * by | token. This method will be used to create the {@link ConfigUtil#PLACEHOLDER_REGEX}.
+     *
+     * @return String that contains placeholder types which are separated by | token.
+     */
+    private static String getPlaceholderString() {
+        StringBuilder stringBuilder = new StringBuilder();
+        for (Placeholder placeholder : Placeholder.values()) {
+            stringBuilder.append(placeholder.getValue()).append("|");
+        }
+        String value = stringBuilder.toString();
```

Comment:
This is fixed in the following commit.

https://github.com/jsdjayanga/carbon-kernel/blob/84257f8c953e5c174ff21f3febb70797228a9d8b/core/src/main/java/org/wso2/carbon/kernel/utils/ConfigUtil.java#L572